.github/actions/aws-ci/action.yml

```diff
@@ -51,15 +51,19 @@ runs:
       run: poetry run pytest -k "aws" --alluredir=./allure/results/ --clean-alluredir
 
     - name: Upload test results
-      if: ${{ success() || failure() }}
+      if: ${{ always() }}
```

Comment:
Several consecutive commits will cancel the workflow run, and in such cases, we don't need to upload test results. So, we can use `!cancelled()` instead of `always()`.

Reply:
It used to be like that (`success() || failure()` == `!cancelled()` ), and I specifically changed it because when we have missing results, then pytest retries, and it gets canceled, so the job is marked as canceled. 

In this case, we want the logs. That's why I changed it.

Reply:
I'm not entirely clear on the scenario. Are you referring to situations where pytest is running and the results are incomplete when the job gets canceled? If the job is interrupted or canceled, I'm not convinced we need to retain the incomplete results.

Reply:
Lets say that because of some bug in code or testcase, or cloud resource configuration, we end up with some expected results to be missing.

Pytest retries trying to find the expected rule with the expected status. 

In that case, the job's timeout might be reached before the pytest ends with an exit code that indicates an error. 

If this happens, we must upload the test results to identify which test case didn't pass. 

I don't remember the specific CI that this happened I could search. 


Perhaps we could better sync pytest timeout vs job timeout?

Reply:
Yes, unfortunately, it introduces a bit of noise and a small delay on consecutive commits. 😞 

Let's see if it becomes a bummer we can revert back to `not canceled`.